contracts/cw1-whitelist-ng/src/contract.rs

```diff
@@ -22,12 +16,12 @@ pub fn validate_admins(api: &dyn Api, admins: &[String]) -> StdResult<Vec<Addr>>
     admins.iter().map(|addr| api.addr_validate(addr)).collect()
 }
 
+#[cw_derive::contract(module=msg)]
 impl<T> Cw1WhitelistContract<T> {
+    #[msg(instantiate)]
     pub fn instantiate(
         &self,
-        deps: DepsMut,
-        _env: Env,
-        _info: MessageInfo,
+        (deps, _env, _info): (DepsMut, Env, MessageInfo),
```

Comment:
This structure is fine, but why not define something like `type MutContext = (DepsMut, Env, MessageInfo)` to make this easier to write?

Reply:
But then I guess that opens the "naming" issue.

More suggestion for future cleanup

Reply:
This argument has to implement `From<(DepsMut, Env, MessageInfo)` - this is the only requirement on this. You can define as you want, as long as it is makeable from such a tuple.

I want to deliver types to be used here wrapping those types for each message type, but it is only QoL, need to first think how to structure them. But it can be easily any user defined wrapper type right now.